src/client/components/Modal/Modal.component.js

```diff
@@ -0,0 +1,83 @@
+/* eslint-disable no-param-reassign */
+import React from 'react';
+import './Modal.styles.css';
+import PropTypes from 'prop-types';
+import Button from '../Button/Button.component';
+import addedToCart from '../../assets/images/added_to_cart.png';
+
+export default function Modal({
+  modalIsOpen,
+  message,
+  cornerCloseButton,
+  primaryButtonLabel,
+  showConfirmIcon,
+  buttonFunction,
+  secondaryButtonFunction,
+  secondaryButtonLabel,
+}) {
+  return (
+    <div className="modal-background">
+      <div className="modal-box">
+        {cornerCloseButton && (
```

Comment:
This prop is thinking too much about the details of the component.
From the outside we don't care if it's a button. What we do care about is if the modal is closable.

Try to rename the props so they make sense from outside, not from inside.